WorldTrotter/ConversionViewController.swift

```diff
@@ -26,47 +27,66 @@ class ConversionViewController: UIViewController{ //, UITextFieldDelegate {
         let existingTextHasDecimalSeparator = textField.text?.range(of: ".")
         let replacementTextHasDecimalSeparator = string.range(of: ".")
         
-        if existingTextHasDecimalSeparator != nil && replacementTextHasDecimalSeparator != nil {
+        //kinda done the way you asked (maybe)
+        let replacementTextHasLetters = string.rangeOfCharacter(from: CharacterSet.letters)
+        
+        if replacementTextHasLetters != nil || existingTextHasDecimalSeparator != nil && replacementTextHasDecimalSeparator != nil{
             return false
         } else {
             return true
         }
+        
+        
     }
     // DELEGATE METHOD : textFieldDidBeginEditing - is called when the user selects the text field
     // TODO: Add and modify the method to build expectation for the output by changing the celsiusLabel when the input field is selected
     // modify the celsiusLabel text to be a single question mark
     // modify the celsiusLabel color to be 60% red, 60% green, and 40% blue (refer to the Developer Documentation for UIColor)
 
+    func textFieldDidBeginEditing(_ textField: UITextField){
+        
+    }
     
     // EVENT HANDLER METHOD : Called when TextField is Changed (notice the optional binding)
-    @IBAction func fahrenheitFieldEditingChanged(_ textField: UITextField) {
-        if let text = textField.text, let value = Double(text) {
+    @IBAction func fahrenheitFieldEditingChanged(_ fahrenheitTextField: UITextField) {
+        if let text = fahrenheitTextField.text, let value = Double(text) {
             fahrenheitValue = Measurement(value: value, unit: .fahrenheit)
         } else {
             fahrenheitValue = nil
         }
     }
+    
+    @IBAction func celsiusFieldEditingChanged(_ celsiusTextField: UITextField){
+        if let ctext = celsiusTextField.text, let v = Double(ctext){
+            celsiusValue = Measurement(value: v, unit: .celsius)
+        } else {
+            celsiusValue = nil
+        }
+    }
     //Stored Properties for Fahrenheit Temperature Measurement w/Observer
     var fahrenheitValue: Measurement<UnitTemperature>? {
         didSet { // this property observer will run after the property is assigned a value
-            updateCelsiusLabel()
+            updateTemp()
         }
     }
     //Computed Property for Celsius Temperature Measurement (Read only property - getter without setter)
     var celsiusValue: Measurement<UnitTemperature>? {
-        if let fahrenheitValue = fahrenheitValue {
-            return fahrenheitValue.converted(to: .celsius)
-        } else {
-            return nil
+        didSet{
+            updateTemp()
```

Comment:
Good use of the property observers; however, call a different function or pass a parameter to identify which property was changed.